cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgresSqlStructuredTable.java

```diff
@@ -502,6 +545,35 @@ private void setField(PreparedStatement statement, Field field,
     }
   }
 
+  /**
+   * Sets the {@link PreparedStatement} arguments by the key {@link Range}.
+   */
+  private void setStatementFieldByRange(Range keyRange,
+                                        PreparedStatement statement) throws SQLException, InvalidFieldException {
+    setStatementFieldByRange(keyRange, statement, 1);
+  }
+
+  /**
+   * Sets the {@link PreparedStatement} arguments by the key {@link Range} starting from the given argument index.
+   *
+   * @return the next argument index that have been set up to
+   */
+  private int setStatementFieldByRange(Range keyRange, PreparedStatement statement,
```

Comment:
This method is called for index = 1. why dont we merge it with the method above or just have this method and remove above method?

Reply:
This method is also called with index != 1 for the multiScan case. The above method is for convenience for basic case.